lib/react.js

```diff
@@ -250,6 +250,12 @@ declare module react {
   declare export var ConcurrentMode: ({children: ?React$Node}) => React$Node; // 16.7+
   declare export var StrictMode: ({children: ?React$Node}) => React$Node;
 
+  declare export var Suspense: React$ComponentType<{
+    children?: ?React$Node,
```

Comment:
Is this needed?
IIUC, this says that it may or may not have children, and they have to be react nodes. [I think that's the default](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoAxnAdgZwC5gBKApgIboEC8YATsQI4CuAlnQBQDkd5eHAlAG5UURlgrNsYAMJwAtgAdsxLHjYAPAFxgA3gF8+WkjwB0AOTgATYjtrE8jGljBZGMGALC6M2fGAD6pGDUADwyCkoqYMAAfEKYuAR+AEZBYKFyiljKeNHBFswAbtHaAEQAFsRucCW6wcD5RXVhmdmxQA).

Reply:
whoa, that seems like a bug – composite components should only take children if it's explicitly specified (it's only host components like `'div'` that always take React$Node children)

Reply:
Based on the context provider and consumer types above, I assumed an explicit `children` prop was needed.

Reply:
FWIW, children is the part of our react model that I'm still not totally familiar with. I'm glad we're having this conversation because it's forcing me to look into it more.

@sophiebits: I'll have to do some more investigation. Requiring an explicit children prop would be an extremely disruptive change at this point.

@bvaughn: I just took a look at our react docs to see what we say about children types. It looks like we have conflicting advice [here](https://flow.org/en/docs/react/types/#toc-react-node) and [here](https://flow.org/en/docs/react/children/). I think the `?React.Node` bit is necessary, but I'm unclear about whether or not the `?children` part is also necessary. I'll have to look into it more, but either way I'm ok with what you have here.

Reply:
@sophiebits: Width subtyping allows extra properties to be passed into an object, so if the Props type is _not_ exact then we allow children even if they are not specified explicitly. On the other hand, if the Props type _is_ exact then [we don't allow the extra properties](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoAxnAdgZwC5gBKApgIboEC8YATsQI4CuAlnQBQDkd5eHAlAG5UURlgrNsYAMJwAtgAdsxLHjYAPAFxgA3gB9dAXz5aSPAHQA5OABNiO2sTyMaWMFkYwYAsAYzZ8YAD6pGDUADwyCkoqYMAAfEKYuASBAEahYBFyiljKeHFh1swAbnHaAEQAFsSecOUGYcBFpY2ROXkJQA).

Mystery solved!

Reply:
for posterity: React's treatment of `children`:

1. React.createElement takes varargs that become `props.children` (JSX children syntax turns into this)
2. host component (`'div'` etc) `children` prop needs to be react nodes and React recurses into them
3. composite component `children` are *not* special in any way; by convention people often use it to contain React nodes then render `<div>{props.children}</div>` or similar, but they can also be other data types (eg: the "render props" pattern often has `children` be a function)